### PR TITLE
IA-368

### PR DESCRIPTION
**Pull request by joshbruhin**
Refactor scrolling to the selected textField when it begins editing so we can scroll to the 'activeTextField' after the keyboard animation. Doing this because the tableView.scrollToRow() function need to be called after the keyboard animation completes because the tableView bounds changes during that animation.

Support Elevate PR: https://github.com/Sage-Bionetworks/elevateMS/pull/203

### CODE REVIEW

File: BridgeAppSDK/SBAGenericStepViewController.swift [hunk 1]
```diff
@@ -682,11 +683,12 @@ open class SBAGenericStepViewController: ORKStepViewController, UITableViewDataS
                 
                 let answerFormat = tableItem?.formItem?.answerFormat?.implied()
                 let type = answerFormat?.questionType
-
+                
                 // set keyboard type
                 if let textAnswerFormat = answerFormat as? ORKTextAnswerFormat {
                     // use the keyboard type defined for this step
                     fieldCell.textField.keyboardType = textAnswerFormat.keyboardType
+                    fieldCell.textField.isSecureTextEntry = textAnswerFormat.isSecureTextEntry
```
Comment (syoung-smallwisdom) [inline BridgeAppSDK/SBAGenericStepViewController.swift]: Should include all functionality of the text answer format:

````
/**
 The autocapitalization type that applies to the user's input.
 By default, the value of this property is `UITextAutocapitalizationTypeSentences`.
 */
@property UITextAutocapitalizationType autocapitalizationType;

/**
 The autocorrection type that applies to the user's input.
 By default, the value of this property is `UITextAutocorrectionTypeDefault`.
 */
@property UITextAutocorrectionType autocorrectionType;

/**
 The spell checking type that applies to the user's input.
 By default, the value of this property is `UITextSpellCheckingTypeDefault`.
 */
@property UITextSpellCheckingType spellCheckingType;

/**
 The keyboard type that applies to the user's input.
 By default, the value of this property is `UIKeyboardTypeDefault`.
 */
@property UIKeyboardType keyboardType;

/**
 Identifies whether the text object should hide the text being entered.
 
 By default, the value of this property is NO.
 */
@property(nonatomic,getter=isSecureTextEntry) BOOL secureTextEntry;
````

File: BridgeAppSDK/SBAGenericStepViewController.swift [hunk 2]
```diff
@@ -778,18 +780,32 @@ open class SBAGenericStepViewController: ORKStepViewController, UITableViewDataS
...
-        guard let customField = textField as? SBAStepTextField else { return }
+        activeTextField = textField
+        scroll(to: textField)
         
-        savedVerticalScrollOffet = tableView!.contentOffset.y
-        tableView?.scrollToRow(at: customField.indexPath!, at: .bottom, animated: true)
+//        guard let customField = textField as? SBAStepTextField else { return }
+//        
+//        savedVerticalScrollOffet = tableView!.contentOffset.y
+//        tableView?.scrollToRow(at: customField.indexPath!, at: .middle, animated: true)
```
Comment (syoung-smallwisdom) [inline BridgeAppSDK/SBAGenericStepViewController.swift]: commented out code. please remove.

**Review by syoung-smallwisdom — CHANGES_REQUESTED** [review]
a few nits.

**Comment by joshbruhin** [comment]
@syoung-smallwisdom done

**Review by syoung-smallwisdom — APPROVED** [review]

**Pull request merged**